presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java

```diff
@@ -605,6 +639,33 @@ public PagesResponse handle(Request request, Response response)
             }
         }
 
+        private void verifyChecksum(URI uri, long readChecksum, List<SerializedPage> pages)
+        {
+            // TODO identify our address, so that query failure reports both ends of communication
+            switch (dataIntegrityVerification) {
+                case NONE:
+                    if (readChecksum != NO_CHECKSUM) {
+                        throw new IllegalStateException(format("Expected checksum to be NO_CHECKSUM (0x%08x) but is 0x%08x, when reading from %s", NO_CHECKSUM, readChecksum, uri));
```

Comment:
How does this surface to the user? It'd be ideal if it could be mapped to a proper error code indicating there's a misconfiguration in the cluster (and skip the retry).

Reply:
The same way as a checksum verification failure, as in https://github.com/prestosql/presto/pull/3438#issuecomment-617750235